### PR TITLE
Encapsulate StoreKit inside MobilePayKit

### PR DESCRIPTION
**Pull request by momo-ozawa**
## Description

This PR encapsulates `StoreKit` completely inside the `MobilePayKit` library. This means that any app using `MobilePayKit` won't need to directly interface with `StoreKit`.

Ref: https://github.com/Automattic/MobilePay-Apple/pull/1#discussion_r672486337

## How to test

1. Go to Edit Scheme > Run > Options > StoreKit Configuration
2. Choose `Configuration.storekit`
3. Run the iOS app
4. Tap on Rocket Fuel to purchase
5. Tap Confirm
6. Tap OK
7. On Xcode debugger menu, click on the Manage StoreKit Transactions menu item (See example below)

Screen Shot 2021-07-19 at 16 29 19

8. ✅ There should be a record of Rocket Fuel being purchased (See example below)

Screen Shot 2021-07-19 at 16 32 32

### CODE REVIEW

File: MobilePayKit/Sources/MobilePayKit/PaymentViewModel.swift [hunk 1]
```diff
@@ -1,12 +1,11 @@
 import Foundation
 import StoreKit
-import MobilePayKit
 
-class PaymentViewModel: NSObject, ObservableObject {
+public class PaymentViewModel: NSObject, ObservableObject {
```
Comment (leandroalonso) [inline MobilePayKit/Sources/MobilePayKit/PaymentViewModel.swift]: Are you planning on adding unit tests for this on a future PR?

Comment (leandroalonso) [reply MobilePayKit/Sources/MobilePayKit/PaymentViewModel.swift]: We don't necessarily need to change that right now but I wonder if we should call it `PaymentViewModel`.

Thiking about the library, we would need an entity that:

1. Return products
2. "Buy" products
3. Return states/errors
4. Can restore a purchase

So maybe naming `PaymentManager` would make sense?

Then, in the app, the user can have a `PaymentViewModel` that connects with this entity.

**Review by leandroalonso — CHANGES_REQUESTED** [review]

File: MobilePayKit/Sources/MobilePayKit/PaymentViewModel.swift (hunk 1 above)
Comment (momo-ozawa) [reply MobilePayKit/Sources/MobilePayKit/PaymentViewModel.swift]: @leandroalonso 

> Are you planning on adding unit tests for this on a future PR?

Yes, definitely! I will make a separate PR with tests.

> Thiking about the library, we would need an entity that:
> 
>     Return products
>     "Buy" products
>     Return states/errors
>     Can restore a purchase
> 
> So maybe naming PaymentManager would make sense?
> 
> Then, in the app, the user can have a PaymentViewModel that connects with this entity.

Yeah, PaymentManager makes sense. I'll make changes to this current PR.

**Comment by momo-ozawa** [comment]
@leandroalonso `PaymentViewModel` is now defined in the app, and is hooked up to MobilePayKit's `PaymentManager`.

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**